reactor-netty-http/src/test/java/reactor/netty/http/client/HttpClientTest.java

```diff
@@ -614,6 +618,79 @@ void sslExchangeRelativeGet() throws SSLException {
 		assertThat(responseString).isEqualTo("hello /foo");
 	}
 
+	@ParameterizedTest
+	@ValueSource(booleans = {true, false})
+	void maxConnectionPools(boolean withMaxConnectionPools) throws SSLException {
+
+		try {
+			ArgumentCaptor<String> argumentCaptor = ArgumentCaptor.forClass(String.class);
+			Logger spyLogger = Mockito.spy(log);
+			Loggers.useCustomLoggers(s -> spyLogger);
+
+			SslContext sslServer = SslContextBuilder
+					.forServer(ssc.certificate(), ssc.privateKey())
+					.build();
+
+			disposableServer = createServer()
+					.secure(ssl -> ssl.sslContext(sslServer))
+					.handle((req, resp) -> resp.sendString(Flux.just("hello ", req.uri())))
+					.bindNow();
+
+			ConnectionProvider connectionProvider = withMaxConnectionPools ? ConnectionProvider
+					.builder("max-connection-pools")
+					.maxConnectionPools(1)
+					.build() : ConnectionProvider
+					.builder("max-connection-pools")
+					.build();
+
+			StepVerifier
+					.create(Flux
+							.range(1, 2)
+							.flatMap(i -> createClient(connectionProvider, disposableServer::address)
+									.secure(ssl -> ssl.sslContext(createClientSslContext()))
+									.get()
+									.uri("/foo")
+									.responseContent()
+									.aggregate()
+									.asString()))
+					.thenConsumeWhile(s -> true)
+					.verifyComplete();
+
+			if (withMaxConnectionPools) {
+				Mockito
+						.verify(spyLogger)
+						.warn(argumentCaptor.capture(), Mockito.eq(2), Mockito.eq(1));
+				assertThat(argumentCaptor.getValue()).isEqualTo(
+						"Connection pool creation limit exceeded: {} pools created, maximum expected is {}");
+			}
+			else {
+				Mockito
+						.verify(spyLogger, times(0))
+						.warn(Mockito.eq(
+										"Connection pool creation limit exceeded: {} pools created, maximum expected is {}"),
+								Mockito.eq(2),
+								Mockito.eq(1));
+
+			}
+		}
+		finally {
+			disposableServer.dispose();
```

Comment:
Let's dispose the `ConnectionProvider` here.
No need to dispose the server it will be disposed by `reactor.netty.BaseHttpTest#disposeServer`